catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/app/UserApplication.java

```diff
@@ -164,6 +200,44 @@ private Map getSubjectPreferences(Subject subject) {
     return Collections.emptyMap();
   }
 
+  private List getSubjectNotifications(Subject subject) {
+    String userid = subjectIdentity.getUniqueIdentifier(subject);
+    try {
+      String filter = String.format("user = '%s'", userid);
+      List<Map<String, Object>> notificationsList =
+          persistentStore.get(PersistenceType.NOTIFICATION_TYPE.toString(), filter, 0, 999);
+      return notificationsList.stream().map(this::mapAttributes).collect(Collectors.toList());
+    } catch (PersistenceException e) {
+      LOGGER.info(
+          "PersistenceException while trying to retrieve persisted notifications for user {}",
+          userid,
+          e);
+    }
+    return Collections.emptyList();
+  }
+
+  private Map mapAttributes(Map<String, Object> persistentItem) {
+    Object srcs = persistentItem.getOrDefault("src_txt", Collections.emptySet());
```

Comment:
❓ Should we strip the stuffixes so we don't need to use them? I.e. call `PersistentItem.stripSuffix` then call `.getOrDefault("src", ...);` instead?

Reply:
I didn't know about that method, perfect